pandas/core/reshape/melt.py

```diff
@@ -24,6 +25,10 @@
 def melt(frame, id_vars=None, value_vars=None, var_name=None,
          value_name='value', col_level=None):
     # TODO: what about the existing index?
+    if isinstance(frame.columns, ABCMultiIndex):
```

Comment:
I'm not especially familiar with melt and multi-index columns, but I don't think this is quite right.

It seems like you need to specify `col_level` when you have a MI in the columns, so you should probably just be checks against `frame.columns.levels[col_level]` when you have a MI.

However, it doesn't quite seem that a `col_level` is *required* when there's a MI in the columns. The default of `pd.melt(df)` seems to work, but any time I specified an `id_vars` or `value_vars` without col_level I get an uninformative error message. I'm not sure what's going on.

Reply:
I think you need to provide `col_level` for MI when only melting on one level like this example from the docstring (that I added a new test for):

`pd.melt(df, col_level=0, id_vars=['A'], value_vars=['B'])`

But you don't need to specify `col_level` when using all levels of MI:
`pd.melt(df, id_vars=[('A', 'D')], value_vars=[('B', 'E')])`

Reply:
All I am doing at L28 is gathering column names from all levels. There are other checks to make sure that melting is performed properly, this will just check to make sure that whatever you pass, it is in your df

Reply:
Thoughts?

Reply:
yeah i think this is ok, can you provdie a comment on what is going on.